openbsd/efivars.md

```diff
@@ -0,0 +1,150 @@
+# Instructions for getting user-space EFI vars API on OpenBSD
+
+Read [README.md](./README.md) first for general information.
+
+This is very similar to what's described in [esrt.md](./esrt.md) and
+[esrt-api.md](./esrt-api.md), but duplication is intentional for simplicity.
+
+The plan is:
+ * obtain `efivars-api` branch of https://github.com/3mdeb/openbsd-src fork of
+   OpenBSD source tree
+ * compile kernel
+ * compile EFI bootloader
+ * install both of them and create `/dev/efi`
+ * compile and install test tool that uses EFI vars API
+ * reboot and check that it works
+
+## Preparation
+
+As a regular user assuming `doas` is configured:
+
+```
+# install git package
+doas pkg_add git
+
+# add your user to wsrc and wobj groups (relogin to apply these changes)
+doas user mod -G wsrc,wobj $USER
+
+git clone --depth 1 -b efivars-api https://github.com/3mdeb/openbsd-src /usr/src
+```
+
+## Kernel compilation
+
+```
+# /sys/ is a symbolic link to /usr/src/sys/, go there to setup kernel build
+cd /sys/arch/amd64/conf/
+
+# generate configuration (the file is already there on the branch)
+config CUSTOM
+
+# build the kernel for the first time
+make -C ../compile/CUSTOM -j$(nproc)
+```
+
+## EFI bootloader compilation
+
+```
+make -C /sys/arch/amd64/stand/efiboot/bootx64
+```
+
+## Installation
+
+```
+# copy the kernel image into root for convenience
+doas cp /sys/arch/amd64/compile/CUSTOM/obj/bsd /bsd.custom
+
+# mount EFI partition
+doas mount -t msdos /dev/sd0i /mnt/
+
+# install EFI bootloader
+doas cp /sys/arch/amd64/stand/efiboot/bootx64/BOOTX64.EFI /mnt/efi/BOOT/
+
+# unmount EFI partition
+doas umount /mnt/
+
+# create /dev/efi file
+doas sh -c 'cd /dev && /usr/src/etc/etc.amd64/MAKEDEV efi'
+```
+
+## Prepare user-space tool
+
+```
+# build sample tool
+make -C /usr/src/usr.sbin/efivartest
+
+# install it
+doas make -C /usr/src/usr.sbin/efivartest install
+```
+
+## Reboot
+
+```
+doas reboot
+```
+
+When bootloader asks for input (`boot>` prompt), type in `bsd.custom` and press
+Enter key.
+
+## Use the tool
+
+### Adding/updating a variable
+
+```
+doas efivartest set aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee test newval
```

Comment:
@SergiiDmytruk At this point I'm getting: `Failed to set var: Operation not permitted`

Reply:
kdump log if it can help to identify problem: https://paste.dasharo.com/?4380c622a7989b8d#tQVkSMbcfcAEAhAYXoqy6tQA9S7SovEmvQVjrPYJQLa

Reply:
It's because of security level. I forgot to mention changing it in the instructions. Added a commit to address this.

Reply:
Yes, that was it of course